pyswagger/io.py

```diff
@@ -325,15 +337,18 @@ def apply_with(self, status=None, raw=None, header=None):
 
             if r and r.schema:
                 # update data from Opeartion if succeed else from responseMessage.responseModel
-                self.__data = r.schema._prim_(self.raw, self.__op._prim_factory)
-
-        if header != None:
-            if isinstance(header, (collections.Mapping, collections.MutableMapping)):
-                for k, v in six.iteritems(header):
-                    self._convert_header(r, k, v)
-            else:
-                for k, v in header:
-                    self._convert_header(r, k, v)
+                content_type = 'application/json'
+                for k, v in six.iteritems(self.header):
+                    if k.lower() == 'content-type':
+                        content_type = v[0]
+                        break
+                if content_type == 'application/json':
```

Comment:
Here is tricky, I don't sure how to handle it in a better way. Actually, I find prim_factory and mime_codec has some similar work.